TykTechnologies/tyk-operator
# TT 3695 - Fix Security Policy Migration

## Description

The migration steps described in [this documentation](https://github.com/TykTechnologies/tyk-operator/blob/v0.12.0/docs/policies/migration.md) do not work as expected. Setting the `id` field in the CR to the `_id` field of the existing Policy has some issues.

## Related Issue




Resolves #204
https://tyktech.atlassian.net/browse/TT-3695

## Motivation and Context


Once you migrate the existing policy from Dashboard to k8s, k8s state should include details of the spec that exists on Dashboard, and from now on, the source of truth must become k8s. Operator must restore subsequent changes from Dashboard in the next reconciliations.

## Test Coverage For This Change



Added test cases to cover:
- Migration of an existing Security Policy from Dashboard to k8s
- Ensuring k8s is the source of truth.
    - Drifts (updates/deletes) from Dashboard should be restored based on the k8s state in the next reconciliation.
- Creating a simple SecurityPolicy should update ApiDefinition's status if that SecurityPolicy refers to it
    - Ensure that corresponding resources are created on Tyk
- Deleting this SecurityPolicy must update ApiDefinition status.
    - Ensure that the corresponding SecurityPolicy is deleted from Tyk as well

## Screenshots (if appropriate)

## Types of changes


## Checklist
